Content.Shared/GameTicking/SharedGameTicker.cs

```diff
@@ -144,18 +145,37 @@ public TickerJobsAvailableEvent(Dictionary<NetEntity, string> stationNames, Dict
         }
     }
 
-    [Serializable, NetSerializable]
-    public sealed class RoundEndMessageEvent : EntityEventArgs
+    [Serializable, NetSerializable, DataDefinition]
+    public sealed partial class RoundEndMessageEvent : EntityEventArgs
     {
-        [Serializable, NetSerializable]
-        public struct RoundEndPlayerInfo
+        [Serializable, NetSerializable, DataDefinition]
+        public partial struct RoundEndPlayerInfo
         {
+            [DataField]
             public string PlayerOOCName;
+
+            [DataField]
             public string? PlayerICName;
+
+            [DataField, NonSerialized]
+            public NetUserId? PlayerGuid;
+
             public string Role;
+
+            [DataField, NonSerialized]
+            public Array JobPrototypes;
+
+            [DataField, NonSerialized]
+            public Array AntagPrototypes;
```

Comment:
Do not store this as an untyped array type, actually store it as the appropriate type (`string[]` or something).